src/lib/openai/openai.service.ts

```diff
@@ -35,22 +35,28 @@ export type RunnerResult = {
 export type ConfigureStreamResolver = (value: RunnerResult) => void;
 export type ConfigureStreamOptions = { threadId: string };
 
+export type OpenAiOptions = {
+  apiKey: string;
+  assistantId: string;
+  language?: string;
+  model?: string;
+};
+
 // timeout padrão de 3 minutos
 const defaultTimeout = 3 * 60 * 1000;
 export class OpenAiService {
   public openai: OpenAI;
 
-  constructor(
-    private readonly apiKey: string,
-    private readonly assistantId: string
-  ) {
-    this.openai = new OpenAI({ apiKey: this.apiKey });
+  constructor(private readonly options: OpenAiOptions) {
```

Comment:
A verificação das propriedades `apiKey` e `assistantId` no construtor da classe `OpenAiService` é uma boa prática para garantir que os parâmetros necessários estão presentes. No entanto, é importante garantir que essas verificações não introduzam comportamentos inesperados no caso de valores nulos ou vazios que possam ser considerados válidos em outros contextos.
Verificar parâmetros obrigatórios no início da criação de um objeto pode prevenir erros de execução mais tarde, mas deve-se considerar todos os cenários possíveis onde um valor poderia ser considerado válido.